scripts/migration/courses-populate.ts

```diff
@@ -0,0 +1,172 @@
+/* eslint-disable no-console */
+/* eslint-disable no-await-in-loop */
+/**
+ * This script retrieves data from all courses from all available
+ * class rosters and pushes them to the 'courses' collection in
+ * Firebase.
+ *
+ * This script also creates a mapping from courses to the rosters
+ * that the course is available in.
+ *
+ *
+ * This script also creates a mapping from crseId to its
+ * catalogNbr.
+ *
+ *  Motivation for this script as well as the schema
+ * for the 'courses' collection can be found here:
+ * https://tinyurl.com/5n86j4p7
+ */
+
+import fetch from 'node-fetch';
+import { FieldValue } from 'firebase-admin/firestore';
+import {
+  coursesCollection,
+  availableRostersForCourseCollection,
+  crseIdToCatalogNbrCollection,
+} from '../firebase-config';
+import { CourseFullDetail } from '../../src/requirements/types';
+
+/** A helper function to generate a wait promise. Used for cooldown to limit API usage. */
+const wait = (time: number) =>
+  new Promise<void>(resolve => {
+    setTimeout(() => resolve(), time);
+  });
+
+/* Retrieves the rosters available on the class roster API */
+const retrieveAvailableRosters = async () => {
+  const rosters: readonly string[] = await fetch(
+    'https://classes.cornell.edu/api/2.0/config/rosters.json'
```

Comment:
Could we maybe refactor this into some constant like we do here? https://github.com/cornell-dti/course-plan/blob/0dba33372dd28b7e2f4d7ed30a3e80ef0e98aa88/src/requirements/courses-json-generator.ts#L9

Reply:
Yeah, will do.